tms/src/main/java/com/netcracker/edu/tms/service/mail/MailServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.netcracker.edu.tms.service.mail;
+
+import com.netcracker.edu.tms.model.Mail;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+import org.springframework.stereotype.Service;
+
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@Service
+public class MailServiceImpl implements MailService {
+
+    private Session session = null;
+    @Value("${notification.mail.host}")
+    private String host;
+    @Value("${notification.mail.password}")
+    private String password;
+    @Value("${notification.mail.threads}")
+    private String threads = "1";
+    private ExecutorService executor = Executors.newFixedThreadPool(Integer.valueOf(this.threads),
```

Comment:
private ExecutorService executor;

@PostConstruct
private void init() {
    executor = Executors.newFixedThreadPool(Integer.valueOf(this.threads),
        new CustomizableThreadFactory("MailSender executor- %d "));
}